### PR TITLE
Remove unnecessary closures

### PR DESCRIPTION
**Pull request by B3rn475**
Some widgets are using closures even if the only values that are
captured are this, context or widget, that can be accessed even from
methods of the State object.

### CODE REVIEW

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 1]
```diff
@@ -278,10 +278,7 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
           children.add(
             new Expanded(
               child: new InkResponse(
-                onTap: () {
-                  if (widget.onTap != null)
-                    widget.onTap(i);
-                },
+                onTap: widget.onTap != null ? () { widget.onTap(i); } : null,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart]: This isn't equivalent, now if you don't provide an onTap, the button won't react. That's probably fine, but we should make sure we document that.

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 2]
```diff
@@ -179,10 +179,15 @@ class _ModalBottomSheet<T> extends StatefulWidget {
 }
 
 class _ModalBottomSheetState<T> extends State<_ModalBottomSheet<T>> {
+
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_sheet.dart]: nit: no blank line here

**Comment by Hixie** [comment]
It's a pity these somewhat lower the readability of the code, but yeah, this seems reasonable.
LGTM

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart (hunk 1 above)
Comment (B3rn475) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart]: Probably better to avoid changes in the behavior

**Pull request merged**